### PR TITLE
DOC: Miscellaneous documentation improvements (part 3)

### PR DESCRIPTION
**Pull request by jhlegarreta**
- DOC: Fix typos and grammar
- DOC: Format properly returned parameter docstrings
- DOC: Make the parameter type dwell on a single line
- DOC: Indent appropriately a parameter docstring
- DOC: Enclose inline literal in double bacticks
- DOC: Add description to parameter docstring
- DOC: Use the appropriate indentation in docstrings
- DOC: Add parameter name/type line to return docstring
- DOC: Add an empty line before list items
- DOC: Indent appropriately enumerated lists

### CODE REVIEW

**Review by skoudoro — COMMENTED** [review]
Hi @jhlegarreta,

It is almost ready to go. Can you address these 2 small comments. Thanks!

File: dipy/align/imaffine.py [hunk 1]
```diff
@@ -429,8 +429,7 @@ def transform(
 
         Returns
         -------
-        transformed : array, shape `sampling_grid_shape` or
-                      `self.codomain_shape`
+        transformed : array, shape `sampling_grid_shape` or `self.codomain_shape`
```
[the remark this answers is not in the document]
Comment (jhlegarreta) [reply dipy/align/imaffine.py]: :+1: Done. Push-forced.

File: dipy/align/imaffine.py [hunk 2]
```diff
@@ -490,8 +489,7 @@ def transform_inverse(
...
```
[the remark this answers is not in the document]
Comment (jhlegarreta) [reply dipy/align/imaffine.py]: :+1: Done. Push-forced.

**Review by skoudoro — APPROVED** [review]
All good, merging.

Thanks @jhlegarreta

**Pull request merged**